### DISCUSSION TITLE
redlining-41 not sure how to proceed

### DISCUSSION THREAD

**Discussion by byandell** [earthlab-education/Earth-Analytics-AY24#623] [Q&A]
Instructions for redlining-41 say

1. Replace `gdf` with your redlining `GeoDataFrame`.
2. Add code to put your `GeoDataFrame` in the same CRS as your raster data.
3. Replace `x_coord` and `y_coord` with the x and y coordinates from your raster data.

Not sure where how to get CRS for raster data, but I tried

```
redlining_plot_gdf = denver_redlining_gdf.to_crs(ccrs.Mercator())
ndvi_plot_da = denver_ndvi_da.rio.reproject(ccrs.Mercator())
band_plot_dict = {
    name: da.rio.reproject(ccrs.Mercator())
    for name, da in denver_merged_das.items()
}
```

That seemed to work. Now not sure how to replace `x_coord` and `y_coord`. Do these come from the range of
`denver_merged_das`? How do I get that?

**Comment by annaposlednik** [comment]
Along the same vein, I'm following the video for redlining-41 and ran into this error with "ValueError: could not convert string to float: 'Denver'" 
![Screenshot 2024-11-15 140501](https://github.com/user-attachments/assets/d2539766-c62a-44fa-954d-0bd02b20c54c)

**Reply by annaposlednik** [reply]
Okay resolved my issue by using "ndvi_da" instead of "denver_ndvi_da"

**Reply by eculler** [reply]
@annaposlednik nice work resolving this on your own! Make sure when you're posting to put all of the relevant code, and full error messages (copying and pasting works better than screenshots) -- I don't think I would have been able to give you much information here.

**Comment by eculler** [comment]
@byandell -- to get the CRS for a `DataArray` (at least, one imported with rioxarray that has a CRS), you use:

```python
da.rio.crs
```

I'd recommend avoiding re-projecting your raster if you can, but it shouldn't affect the zonal stats too much. Still, the Mercator projection is pretty notoriously inaccurate so be careful with that. We use it a lot for web mapping because the basemap tiles come in it, but otherwise I try to steer clear.

To get the x and y coordinates, assuming they are named `x` and `y`, you can access them directly from the `DataArray`, kind of like we do for columns in a `DataFrame`:

```python
x_coord = da.x
y_coord = da.y
```
